sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -145,6 +157,28 @@ class device_image : public detail::device_image_plain,
     return device_image_plain::has_kernel(KernelID, Dev);
   }
 
+  backend ext_oneapi_get_backend() const noexcept {
+    return device_image_plain::ext_oneapi_get_backend_impl();
+  }
+
+#if (!defined(_HAS_STD_BYTE) || _HAS_STD_BYTE != 0)
+  template <sycl::bundle_state T = State,
+            typename = std::enable_if_t<T == bundle_state::executable>>
+  std::vector<std::byte> ext_oneapi_get_backend_content() const {
+    return device_image_plain::ext_oneapi_get_backend_content_impl();
+  }
+
+#ifdef __cpp_lib_span
+  template <sycl::bundle_state T = State,
+            typename = std::enable_if_t<T == bundle_state::executable>>
+  std::span<std::byte> ext_oneapi_get_backend_content_view() const {
```

Comment:
@gmlueck 
Unless I'm mistaken, returning `std::span<std::byte>` here allows the user to modify the raw contents in the underlying memory. Is this intended?
One other implementation would be to instead first create a vector copy of the contents as a member of `device_image` and return a span of that object instead. However, the way its written, i think the spec does hint that the span references the raw contents directly and not a copy of theirs.

Reply:
Oh, that's a good catch.  It should return `std::span<const std::byte>` instead.

We definitely don't want to create a vector copy of the data.  The whole point of this API is to give the application access to the data without incurring the overhead of making a copy.

Reply:
If you agree, do you mind updating the spec also as part of this PR?